shell/platform/glfw/flutter_glfw.cc

```diff
@@ -325,15 +339,21 @@ static void SetEventLocationFromCursorPosition(
   glfwGetCursorPos(window, &event_data->x, &event_data->y);
 }
 
-// Set's |event_data|'s phase to either kMove or kHover depending on the current
-// primary mouse button state.
+// Set's |event_data|'s phase depending on the current mouse state.
+// If a kUp or kDown event is triggered while the current state is already
+// up/down, a hover/move will be called instead to avoid a crash in the Flutter 
+// engine.
 static void SetEventPhaseFromCursorButtonState(
     GLFWwindow* window,
-    FlutterPointerEvent* event_data) {
+    FlutterPointerEvent* event_data,
+    int64_t buttons) {
+  auto* controller = GetWindowController(window);
   event_data->phase =
-      glfwGetMouseButton(window, GLFW_MOUSE_BUTTON_LEFT) == GLFW_PRESS
-          ? FlutterPointerPhase::kMove
-          : FlutterPointerPhase::kHover;
+      buttons
```

Comment:
I am confused at the logic here.
if buttons evaluate to true (at lease on button is down) and controller->pointer_currently_down, shouldn't we issue move? It feel to me the condition should be reverted.